doc/languages-frameworks/python.section.md

```diff
@@ -374,6 +374,40 @@ mkPythonMetaPackage {
 }
 ```
 
+#### `mkPythonEditablePackage` function {#mkpythoneditablepackage-function}
+
+When developing Python packages it's common to install packages in [editable mode](https://setuptools.pypa.io/en/latest/userguide/development_mode.html).
+Like `mkPythonMetaPackage` this function exists to create an otherwise empty package, but also containing a pointer to an impure location outside the Nix store that can be changed without rebuilding.
+
+Note that overriding packages deeper in the dependency graph _can_ work, but it's not the primary usecase and overriding existing packages can make others break in unexpected ways.
+
```

Comment:
Thank you for high-quality docs!

I think we should more explicitly mention that our "editable package" always uses hatchlings, either here on in a comment inside the nix source.

The reason to be upfront about this technical detail is that I believe this approach is a very good start and should work for many use-cases around editable installs;
But I have a suspicion that it might not cover all behavior that is allowed via pep518 and it's hooks. Or replicate the functionality of `pip install -e .`, which might or might not be a bad thing in itself.

Reply:
> I think we should more explicitly mention that our "editable package" always uses hatchlings, either here on in a comment inside the nix source.

Added a note, also explaining that build systems becomes runtime dependencies.